### DISCUSSION TITLE
OpenGraph Image issue (mostly with Facebook) in Astroid Open Graph tab in menu

### DISCUSSION THREAD

**Discussion by Brandocracy** [templaza/astroid-framework#844] [Ideas]
I've had some OpenGraph issues with images in the OG Astroid tab for Joomla menu items, found a work around: It seems the Astroid image URL for OG image adds a lot of detail that Facebook at least can't cope with (image size and specific location info).
My work around is to delete the image from the OG tab and in the  Custom Code tab add the following to the "Before </head>" field:  <meta name="og:image" property="og:image" content="FULL OG IMAGE URL" />
Attached screenshot examples.
The issue I think is the image URL in the OG tab looks something like this: OGImageURL#joomlaImage://local-OGImageURL?width=1200&height=630 - and Facebook (et al) doesn't know that to do with that.
![Screenshot 2024-10-25 at 9 10 29 AM](https://github.com/user-attachments/assets/0188d248-fb24-47e7-aae9-022cc134297e)
![Screenshot 2024-10-25 at 9 12 51 AM](https://github.com/user-attachments/assets/b8a83371-79af-4a03-a7f9-81207b006065)

**Comment by paurray** [comment]
Hello @Brandocracy 

I too have had problems sharing to FB and gave up a while ago https://github.com/templaza/astroid-framework/discussions/578 Things worked again for me and then no more...

Maybe it is multiple user error or maybe this is a bug?
One way or the other it is to be honest over my head.
I looked at this again this morning and still cannot get it to work...

First I read the instructions again here:
https://docs.astroidframe.work/menu-item-settings/open-graph-options

I have a vague recollection of there being an Open Graph setting in the Astroid Backend settings.
But if I am not mistaken that has been removed?
Currently on Framework Version: 3.1.6 here.

Any way I tried sharing from 2 sites:
And used the Facebook debug tool, but could not get it to work:
https://developers.facebook.com/tools/debug/

HERE:
https://astroidframe.work/blog/astroid-hands-on/deep-dive-astroid-boilerplate-site-205
and
https://finalbug.tv/en/video-reference-projects/planning-editing-webinars/deep-dive-astroid-boilerplate-site-204-en

Getting the following errors respectively:

The following required properties are missing: fb:app_id
and
No error message as such but also does not work

Any body got any ideas?

**Comment by heikowtb** [comment]
Exactly the same problem here, I too think the problem is the part after "#" in the URL to the pic. 
I wanted to try the workaround, but I am missing the "custom code" tab in the article. Do I have to enable it first? Where?
Regards,
Heiko

**Comment by jazzmang** [comment]
Same issue here: This is what Astroid generated on the frontend and will not work with FB (just blank)
<meta name="og:image" property="og:image" content="https://XYZ.com/images/New-Construction/wilma-new-construction.jpg#joomlaImage://local-images/New-Construction/wilma-new-construction.jpg?width=400&amp;height=300">

The only workaround is a custom code header tag, which does work and displays correctly.
<meta name="og:image" property="og:image" content="https://XYZ.com/images/New-Construction/wilma-new-construction.jpg">
